### PR TITLE
New slack permissions step

### PR DESCRIPTION
**Pull request by thepug**
Test with
https://github.com/open-company/open-company-web/pull/515

### CODE REVIEW

File: src/oc/auth/api/slack.clj [hunk 1]
```diff
@@ -248,23 +257,19 @@
                                                 (assoc :admin (user-res/admin-of conn (:user-id user)))
                                                 (assoc :slack-id (:slack-id slack-user))
                                                 (assoc :slack-token (:slack-token slack-user))) :slack)
-
             ;; Determine where we redirect them to
             bot-only? (and target-team ((set (:slack-orgs target-team)) (:slack-org-id slack-org)))
-            redirect-arg (if bot-only? :bot :team)]
+            redirect-arg (if (and (not bot-only?) slack-org (not (contains? slack-org :bot-token))) :bot :team)]
```
Comment (bago2k4) [inline src/oc/auth/api/slack.clj]: @thepug why `bot-only?` check has become `(not bot-only?)`? I'm didn't check the rest of the logic of this but i think the old `bot-only?` should still be working.
In fact if you signup the first time, skip the bot, then go to settings and add the bot you are redirected there with `access=team` not bot.

**Review by bago2k4 — CHANGES_REQUESTED** [review]
@thepug small issue with the add bot functionality from the settings panel

File: src/oc/auth/api/slack.clj (hunk 1 above)
Comment (thepug) [reply src/oc/auth/api/slack.clj]: Yeah, I was trying to keep the same variable names but maybe that isn't a good idea.  The `bot-only?` wasn't working as I expected.  I'll give it another round with the add to team test.

**Comment by thepug** [comment]
@bago2k4 addressed https://github.com/open-company/open-company-auth/pull/41#pullrequestreview-119936725 with the last commit d794c41

**Comment by thepug** [comment]
[Create a screen asking to continue to Carrot or install the app on Slack](https://trello.com/c/VZAYDEJh/1304-create-a-screen-asking-to-continue-to-carrot-or-install-the-app-on-slack)

**Review by bago2k4 — APPROVED** [review]

**Pull request merged**